src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java

```diff
@@ -223,7 +223,7 @@ public class MatchXpathCheck extends AbstractCheck {
     public static final String MSG_KEY = "matchxpath.match";
 
     /** Specify Xpath query. */
-    private String query = "";
+    private String query;
```

Comment:
somthing is broken in this Check.

if user did setQuery ones he can not return Check to default state by setting `setQuery("")` by providing default value.
it is not possible to set `query` twice in our config, but plugings (sonar, eclipse-cs) are working directly with modules (no xml), so it might be problem for them.

I am ok with pure junit here , but we need to have default and we need to restore Check instance to default mode of execution by setting defalt value in its setter.

Reply:
And we need to reset xpathExpression field if user set empty value.
Empty string should mean null in xpathExpression

Reply:
No issue I will look on this by tommorow

Reply:
@Kevin222004 any update on this?

Reply:
@rdiachenko sorry I haven't visited it yet will update as soon as possible

Reply:
@romani I haven't missed https://github.com/checkstyle/checkstyle/pull/13242#discussion_r1249602479 this but, 
please look at the updated pr. I have make a slight change in setter. I don't think it is required to restore check instance.
 I added all the kind of test https://github.com/checkstyle/checkstyle/pull/13242#discussion_r1249717443 please check

Reply:
Having null as default, make it impossible to set back to default.
There is no null as value in xml 

What I meant is to update setter to:
```
if (!query.isEmpty()) {
....
} else {
this.query = "";
this.xpathExpression = null;
}
```

Reply:
Not sure I'm following the request with null. @Kevin222004 what's the latest status of this PR?

Reply:
We should remove such field, having `this.xpathExpression` is enough.

Reply:
@Kevin222004 , what is problem to remove this field